### PR TITLE
commitlog: Make bitflip test a proptest

### PR DESCRIPTION
**Pull request by kim**
The test sometimes fails. As a proptest, we'll be able to seed it with failing inputs.

Fixes: #1167

# Expected complexity level and risk

1.8

0.8 for the proptest generator

### CODE REVIEW

File: crates/commitlog/src/tests/bitflip.rs [hunk 1]
```diff
@@ -1,73 +1,155 @@
...
+/// Size in bytes of the (dummy) transactions to generate.
+const TX_SIZE: usize = 32;
+/// Number of transactions to generate per commit.
+const TXS_PER_COMMIT: usize = 10;
 
-    const NUM_COMMITS: usize = 100;
-    const TX_SIZE: usize = 32;
-    const TXS_PER_COMMIT: usize = 10;
-    const COMMIT_SIZE: usize = Commit::FRAMING_LEN + (TX_SIZE * TXS_PER_COMMIT);
+/// The size in bytes of one commit acc. to above parameters.
```
[the remark this answers is not in the document]
Comment (gefjon) [reply crates/commitlog/src/tests/bitflip.rs:31]: ☝️

[the remark this answers is not in the document]
Comment (gefjon) [reply crates/commitlog/src/tests/bitflip.rs:31]: Now you know how I feel all the time!

[the remark this answers is not in the document]
Comment (kim) [reply crates/commitlog/src/tests/bitflip.rs:31]: Are you saying that “acc. to” is not in the same category as “etc.”, “wrt”, “iff”, “wtf”?

File: crates/commitlog/src/tests/bitflip.rs [hunk 2]
```diff
@@ -1,73 +1,155 @@
...
+            .field("bit_mask", &self.bit_mask)
+            .field("segment_offset", &self.segment_offset)
+            .finish()
+    }
+}
+
+impl Inputs {
+    fn generate() -> impl Strategy<Value = Inputs> {
+        // hey proptest, `prop_compose` doesn't make this any more pleasant.
+        // how about, you know, do-notation?
```
[the remark this answers is not in the document]
Comment (kim) [reply crates/commitlog/src/tests/bitflip.rs:71]: Infuriating! :shakes_fist:

**Pull request merged**